### ISSUE TITLE
technical problem with form_count_dof function in proxy

### ISSUE THREAD

**Issue by ricardoavelino** [BlockResearchGroup/compas_ags#31]
Hi @brgcode,

I noted that within the package there are two different ways to take information about the fixed vertices. This causes issues when computing the equilibrium matrix using the method `equilibrium_matrix(...)` from _compas.numerical_. 

For part of the current code the leaf-nodes are taken by default as fixed vertices. For part of the code they are not. Here are some examples of methods considering the fixed vertices in different ways:

- Considering as fixed only the vertices with attribute '_is_fixed_' explicitly set to _True_:

> `def form_identify_dof(...)` (at)  _compas_ags.ags.graphicstatics_
> `def form_count_dof(...)` (at)  _compas_ags.ags.graphicstatics

- Considering as fixed the vertices corresponding to leaves:

> `def form_update_q_from_qind(...)` (at)  _compas_ags.ags.graphicstatics_
> `def compute_jacobian(...)` (at)  _compas_ags.ags2.rootfinding_
> `def compute_form_from_force_newton(...)` (at)  _compas_ags.ags2.rootfinding_,
...and others specially in  _compas_ags.ags2_

I suggest that we keep by default the second "fixity" check, because it is intuitive and helps to do the selection on the right number of independent edges from the start. Plus, on the ags type of input (2D structures drawn with leaves) the first fixity check does not seem useful, does it?

Thanks.

**Comment by ricardoavelino** [comment]
This is working smooth now. But me and @duchaoyu still have issues when using the following function in proxy.

`def form_count_dof(...) `

It trows an error when it is called with the proxy even if it still works outside rhino. @brgcode  Can you help us in that?

**Comment by ricardoavelino** [comment]
This is the error, when running it from a command... But It also does not work for a simple script.

```
Exception Occurred 
Message: Fault

Traceback:
  line unknown, in <module>, "<string>"
  line 800, in close, "/Applications/Rhinoceros.app/Contents/Frameworks/RhCore.framework/Versions/Current/Resources/ManagedPlugIns/RhinoDLR_Python.rhp/Lib/xmlrpclib.py"
  line 1493, in parse_response, "/Applications/Rhinoceros.app/Contents/Frameworks/RhCore.framework/Versions/Current/Resources/ManagedPlugIns/RhinoDLR_Python.rhp/Lib/xmlrpclib.py"
  line 1316, in single_request, "/Applications/Rhinoceros.app/Contents/Frameworks/RhCore.framework/Versions/Current/Resources/ManagedPlugIns/RhinoDLR_Python.rhp/Lib/xmlrpclib.py"
  line 1283, in request, "/Applications/Rhinoceros.app/Contents/Frameworks/RhCore.framework/Versions/Current/Resources/ManagedPlugIns/RhinoDLR_Python.rhp/Lib/xmlrpclib.py"
  line 1598, in _ServerProxy__request, "/Applications/Rhinoceros.app/Contents/Frameworks/RhCore.framework/Versions/Current/Resources/ManagedPlugIns/RhinoDLR_Python.rhp/Lib/xmlrpclib.py"
  line 1243, in __call__, "/Applications/Rhinoceros.app/Contents/Frameworks/RhCore.framework/Versions/Current/Resources/ManagedPlugIns/RhinoDLR_Python.rhp/Lib/xmlrpclib.py"
  line 318, in _proxy, "/Applications/Rhinoceros.app/Contents/Frameworks/RhCore.framework/Versions/Current/Resources/ManagedPlugIns/RhinoDLR_Python.rhp/Lib/compas/rpc/proxy.py"
  line 27, in RunCommand, "/Users/mricardo/Library/Application Support/McNeel/Rhinoceros/6.0/Plug-ins/PythonPlugins/AGS{6971119f-8b5a-4cd2-aad8-edad37a243c9}/dev/AGS_check_equilibrium_cmd.py"
  line 1, in <module>, "<string>"
```

Can this be related to the fact that the function `form_count_dof(...)` returns more than one argument? [k, m] instead of a form.data or force.data and this breaks the proxy?

**Comment by brgcode** [comment]
problem solved in 89c1d103
the return data type was a tuple of `numpy.int64` which can't be converted to a JSON compatible format.
i added conversion to `int`...

however, there are two separate problems in this issue. please, reopen the first part on a separate thread...
